### PR TITLE
Play with Capact

### PR DESCRIPTION
**Pull request by mszostok**
## Description

**Preview URL:**
- ~old:  https://3dc9be72.website-8qa.pages.dev/blog/play-with-capact-via-cli~
- new: https://play-with-capact.website-8qa.pages.dev/blog/take-capact-for-a-spin

Changes proposed in this pull request:

- Add blog post describing "Play with Capact" 
  -  Change diagrams colors to match our website
  - Add diagrams for "dark" theme
  - Use simplified manifests 

> **NOTE:** The Argo UI, was was not described.

## Blog related

**Alternative titles:**  
  - Experiment with Capact  
  - Your first night with Capact   _- could be controversial 👀_ 

---

**Alternative scenario:**

I spot that current examples don't show another important value of Capact: glue different tools and unify it usage via Capact CLI. Because of that, I thought about a different example. 

Currently, in both Implementations we use `echo` to produce just a different string.
Instead, we could  use `helm` and `kubectl` CLIs to produce the ConfigMap manifest.

Implementations:
- First,`helm template hello-world` which produces:
   ```yaml
    # Source: hello-world/templates/config-map.yaml
    apiVersion: v1
    kind: ConfigMap
    metadata:
      name: hello-world
    data:
      message: |
        Hello World from Helm template
   ```
- and second, `kubectl kustomize hello-world` which produces:
  ```yaml
  apiVersion: v1
  data:
    message: |
      Hello World from Kustomize
  kind: ConfigMap
  metadata:
    name: hello-world
  ```

By default, Helm template is used, By using proper Policy, the `kubectl kustomize` is used instead.
```yaml
rules:
  - interface:
      path: cap.interface.capactio.examples.manifest
    oneOf:
    - implementationConstraints:
        attributes:
          - path: cap.attribute.capactio.examples.kustomize
            revision: 0.1.0
```

Pros:
- we can get rid of `kubectl` step, as we will use only the `capact ti get` with `yq`
- showcase that two different popular tools were used transparently

Cons:
- complicates a bit our current example

You can download [hello-world.zip](https://github.com/capactio/website/files/7937263/hello-world.zip) and execute command describe above.

I discussed that with @pkosiec, but the outcome was that we should keep our example as simple as possible (KISS).

## Testing

Populate Hub with manifests from https://github.com/capactio/hub-manifests/pull/53

## Related issue(s)

Fix https://github.com/capactio/website/issues/63

### CODE REVIEW

File: blog/2022-01-19-play-with-capact-via-cli.mdx [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+## What is so special about it?
+
+Our example was simplified to a minimum, but you could see that you can:
+
+- chain different steps together,
+- and more importantly, write once and swap implementation details with ease.
+
+Capact is also great if you want to chain different tools together and keep the same entry point (UX) but that requires more complex examples. For now, we recommend you to take a look at our other examples described in the [**How can I get involved and learn more?**](#how-can-i-get-involved-and-learn-more) section.
+
+## Behind the wall
```
Comment (pkosiec) [inline blog/2022-01-19-play-with-capact-via-cli.mdx:274]: I get the reference, but are you sure it has similar meaning as "Under the hood"? Please consult that with TW or some native Enligsh speaker, if possible 🙂

File: blog/2022-01-19-play-with-capact-via-cli.mdx [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+
+As you saw, you can use CLI to browse Hub manifests, create a specific Action for a given Interface and retrieve the output. CLI communicates with our Gateway via GraphQL calls, same as the UI (Capact Dashboard). You can read more in the [E2E Architecture](/docs/next/architecture/e2e-architecture) document.
+
+However, a more important question is, how did we change the output message?
```
Comment (pkosiec) [inline blog/2022-01-19-play-with-capact-via-cli.mdx:278]:
```suggestion
However, a more important question is, how did it happen?
```

File: blog/2022-01-19-play-with-capact-via-cli.mdx [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+
+:::tip
+Instead of providing the full Interface path, we use **alias** assigned to the imported Interface.
+:::
+
+As a result, our Engine and, in particular, Argo Renderer has an option to put there an Implementation that fulfills the specified Interface. Thanks to that, this step is swappable based on the defined Policy.
+
+## How can I get involved and learn more?
+
+The [Getting Started](/docs/next/getting-started) goes into more detail about how to start with Capact and use its more-advanced features. There are also [Rocket.Chat](/docs/next/example/rocketchat-installation) and [Mattermost](/docs/next/example/mattermost-installation) tutorials that we keep up-to-date for you to walk you through using Capact in real cases.
```
Comment (pkosiec) [inline blog/2022-01-19-play-with-capact-via-cli.mdx:325]: Please search the whole document and remove all `/next` occurrences as we should redirect to the latest stable version. I know that the CI will probably fail, but still - put a TODO somewhere in PR desc or something, to remember about that before merge 🙂 

```suggestion
The [Getting Started](/docs/getting-started) goes into more detail about how to start with Capact and use its more-advanced features. There are also [Rocket.Chat](/docs/example/rocketchat-installation) and [Mattermost](/docs/example/mattermost-installation) tutorials that we keep up-to-date for you to walk you through using Capact in real cases.
```

File: blog/2022-01-19-play-with-capact-via-cli.mdx [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+8. Delete the Action:
+
+   ```bash
+   capact act delete hello
+   ```
+
+Let's consider another example, as life isn’t all sunshine and rainbows. Sometimes we need a more positively-colored message. Capact allows you to do it with ease ✨
+
+To choose a different Implementation for the `say` Interface, we need to prepare a dedicated Policy. The power of it is that you don't have to change the main workflow. You treat the step as a building block and just swap it for another existing one.
+
```
Comment (pkosiec) [inline blog/2022-01-19-play-with-capact-via-cli.mdx:262]: I think that's a great place to put this GIF (or different one) 😄 
tenor-139575536

File: blog/2022-01-19-play-with-capact-via-cli.mdx [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+   capact ti get {ID} -ojson | jq -r '.[0].latestResourceVersion.spec.value.message'
+   ```
+
+8. Delete the Action:
+
+   ```bash
+   capact act delete hello
+   ```
+
+Let's consider another example, as life isn’t all sunshine and rainbows. Sometimes we need a more positively-colored message. Capact allows you to do it with ease ✨
```
Comment (pkosiec) [inline blog/2022-01-19-play-with-capact-via-cli.mdx:259]: It's just my personal opinion, but I would change that to something like:

```suggestion
What if you wanted to output a different message, that reminds you about a masterpiece song you can listen to endlessly? With Capact that's really easy ✨
```

(I'm not convinced to that as well, maybe we can find something else) 😄

**Review by pkosiec — COMMENTED** [review]
A really solid blog post 👌  The tutorial works as expected. Here are my suggestions 🙂

File: blog/2022-01-19-play-with-capact-via-cli.mdx (hunk 4 above)
Comment (mszostok) [reply blog/2022-01-19-play-with-capact-via-cli.mdx:262]: good idea, I also like that one 👍 
Other ideas:
1. no-more-training
2. shocked-happy

I also created dedicated ones:

3. drawing

4. drawing


let's also for others to decide :)

**Comment by mszostok** [comment]
I added 👀  emoji to @pkosiec comments, which will be resolved after brainstorming :)

File: blog/2022-01-19-play-with-capact-via-cli.mdx (hunk 4 above)
Comment (sunny-dasgupta-braingu) [reply blog/2022-01-19-play-with-capact-via-cli.mdx:272]: I like Star Wars meme #4

**Review by sunny-dasgupta-braingu — APPROVED** [review]

File: blog/2022-01-19-play-with-capact-via-cli.mdx [hunk 6]
```diff
@@ -0,0 +1,334 @@
+---
+slug: play-with-capact-via-cli
+title: Play a bit with Capact!
```
[the remark this answers is not in the document]
Comment (tmitchell) [reply blog/2022-01-19-play-with-capact-via-cli.mdx:3]: Maybe "Take Capact for a spin"?

**Review by tmitchell — APPROVED** [review]
Looks great!

**Review by pkosiec — APPROVED** [review]
LGTM

**Pull request merged**